gpii/node_modules/ontologyHandler/src/ontologyHandler.js

```diff
@@ -106,6 +107,111 @@ URLs - this aspect is taken care of in the preferences server component.
+     * Currently assumes that all edges are weight 1.
+     *
+     * @edges (Array) - An array of arrays of length 2, consisting of the named
+     *     edges such as [['A','B'], ['B','C']]
+     * @returns (Object) - An object containing the necessary structures to resolve
+     *     paths using the floydWarshallPath function
+     */
+    gpii.ontologyHandler.floydWarshallAlgorithmWithReconstruction = function ( edges ) {
+        var initialVertices = [];
+        fluid.each(edges, function(edge) {
+            initialVertices.push(edge[0], edge[1]);
+        });
+        var vertices = fluid.arrayToHash(initialVertices);
+        // Assign each vertice an integer key for the lookup arrays
+        fluid.each(Object.keys(vertices), function(vert, i) {
+            vertices[vert] = i;
+        });
+        var numVertices = Object.keys(vertices).length;
+        // Create |V|x|V| dist and next arrays and initialize to Infinity and null
+        var dist = [];
+        var next = [];
+        for (var i = 0; i < numVertices; i++) {
+            dist.push([]);
+            next.push([]);
+            for (var j = 0; j < numVertices; j++) {
+                dist[i].push(Infinity);
+                next[i].push(null);
+            }
+        }
+        fluid.each(edges, function(edge) {
+            var u = vertices[edge[0]], 
+                v = vertices[edge[1]];
+            dist[u][v] = 1; // All weights currently 1
+            next[u][v] = v;
+        });
+        // Standard Floyd-Warshall implementation
+        for (var k = 0; k < numVertices; k++) {
+            for (var i = 0; i < numVertices; i++) {
+                for (var j = 0; j < numVertices; j++) {
+                    if (dist[i][k] + dist[k][j] < dist[i][j]) {
+                        dist[i][j] = dist[i][k] + dist[k][j];
+                        next[i][j] = next[i][k]
+                    }
+                }
+            }
+        }
+        return {
+            vertices: vertices,
+            numVertices: numVertices,
+            dist: dist,
+            next: next
+        }
+    };
+
+    /*
+     * Uses the data generated in floydWarshallAlgorithmWithReconstruction to
+     * generate the shortest path as an array of vertice names.
+     *
+     * @data (Object) - The data structure returned from 
+     *     floydWarshallAlgorithmWithReconstruction
+     * @v1 (String) - The starting vertice
+     * @v2 (String) - The ending vertice
+     * @returns (Array) - An array of vertices for the shortest path, or an 
+     *     empty array if the path does not exist.
+     */
+    gpii.ontologyHandler.floydWarshallPath = function ( data, v1, v2 ) {
+        if (data.vertices[v1] === undefined || data.vertices[v2] === undefined) {
+            return [];
+        }
+        var path = [],
+            u = data.vertices[v1],
+            v = data.vertices[v2];
+        if (data.next[u][v] === null) {
```

Comment:
You're right, the early return can't be skipped since it handles the special case of disconnected vertices. However we can make a few cleanups as below: